flask_platform_reviewofpurchaseitem/forms.py

```diff
@@ -0,0 +1,27 @@
+from flask_platform_components.base_form import BaseForm
+from flask_platform_components import db
+from flask_babelex import lazy_gettext
+from flask_login import current_user
+from .models import ReviewOfPurchaseItem
+
+class ReviewOfPurchaseItemForm(BaseForm):
+    exclude_field_in_model = BaseForm.exclude_field_in_model
+
+    def validate(self, *args, **kwargs):
+        #TODO Add validation for the required specificities
+        if not super(ReviewOfPurchaseItemForm, self).validate(*args, **kwargs):
+            return False        
+        
+        user = self.user.data if current_user.has_role('admin') else current_user
+        
+        # Other than the buyer can't review this purchase item
+        if self.purchase_item.data and self.purchase_item.data.purchase.buyer.id != user.id:
+            self.purchase_item.errors.append(str(lazy_gettext('you cannot make review to this purchase item')),)
+            return False
+
+        # The buyer can't review more than once this purchase item
+        if db.session.query(ReviewOfPurchaseItem).filter(ReviewOfPurchaseItem.purchase_item==self.purchase_item.data, ReviewOfPurchaseItem.user==user).first():
```

Comment:
```suggestion
        if ReviewOfPurchaseItem.query.filter(ReviewOfPurchaseItem.purchase_item==self.purchase_item.data).first():
```
You do not need to specifiy the user as only one review can be done per purchase item.